### ISSUE TITLE
Changing the main windows buttons order and shortcuts.

### ISSUE THREAD

**Issue by Sawuare** [LMMS/lmms#3695]
### New proposal:
"FX Mixer" then "Controller rack" then "Project notes".
### Original proposal:
These main windows buttons:
![capture20170712130032527](https://user-images.githubusercontent.com/25503477/28128677-317ef81e-6739-11e7-917c-0e04ad43ad3f.png)
I think that the last 3 buttons to the left aren't well ordered.
I think that they should be order this way:
"Controller rack" then "FX Mixer" then "Project notes".

And I think re-ordering them should change their shortcuts:
"Controller rack" to F9 (from F11).
"FX Mixer" to F10 (from F9).
"Project notes" to F11 (from F10).

I think my proposed order is better than the current one because:
(1) "Project notes" are usually the last thing to do when working on a project, and are usually used less than the other main windows, so the button and the shortcut of them should be given the least priority, and they should be the last in the order.
(2) The "Controller rack" is more related to the "Automation editor" than the "FX Mixer", so the button and the shortcut of should follow the those of the "Automation editor" and that puts the "FX Mixer" before the "Project notes" which is last.

**Comment by Spekular** [comment]
I would agree on placing project notes last, but I think the FX mixer
should stay before the controller rack as it's a "bigger" feature (you do
more things in it than the controller rack).

**Comment by Sawuare** [comment]
Yes the mixer is a bigger feature and you do more stuff in it, but your audio will be processed by what is in the controller rack, then sent to the mixer, which is the last stage before the audio output, isn't it?

Think of the processing of the audio in the controller rack as part of the audio, which will be processed later in the mixer.

**Comment by Spekular** [comment]
In this case, BB editor would go before song editor, and Automation Editor + Piano roll before that.

**Comment by Sawuare** [comment]
So for the first 4 buttons, you're suggesting this order: automation editor > piano roll > BB E > song editor, right?
To get the BB E you will first need the song editor ;)
I can't understand why you want the automation editor and piano roll in that order.

This is my opinion on the order of the buttons based on streamlined workflow:
I think everyone will agree that the last 4 buttons should be:
controller rack > automation editor > mixer > project notes.
The first 3 buttons are subjective (but the song editor is always first), some users like creating rhythms first, so in this case:
song editor > BB E > piano roll.
And some user like creating melodies and harmonies first, so in this case:
song editor > piano roll > BB E.

I don't care which is first, BB E or piano roll because I don't always use one method over the other, but I think the first order is better because the song editor and BB E buttons share a similar look.

My suggested order:
Song editor > BB E > piano roll > controller rack > automation editor > mixer > project notes.

Sorry if I'm overcomplicating the issue here :)

**Comment by Spekular** [comment]
I don't want Automation->Piano Roll->BB Editor->Song Editor. I'm just
pointing out that if we're ordering the way you suggested we'd have to have
something like that.

You wanted the controller rack to go before the mixer because controllers
affect the audio that goes into the mixer.

Piano roll and Automation editor patterns can be inside a BB track and
therefore affect it. BB tracks affect what goes into the song editor.

I personally think the current order is logical, aside from perhaps the
last two. All the editors are grouped, followed by the mixer. This is where
people will spend most of their time. The controller rack and notes are
very minor windows in comparison, used more rarely, so they come later.
Since projects notes aren't related to actual audio work, unlike the other
windows, it would make sense to move it far right.

>

**Comment by follower** [comment]
FWIW when I saw #3938 it occurred to me that while the right-most icon is "last" (at least in a left-to-right scanning culture) in the list, in terms of proximity to the main work area it's actually the closest icon--which implies importance which seems to be the opposite of what you're wanting to prioritise.

While I have no data to back this supposition up, I suspect the real-world "priority order" that results from icon proximity, ease of identification and muscle memory is probably something more like this: 

```
[ 1 ][ 3 ][ 7 ][ 5 ][ 6 ][ 4 ][ 2 ]
```

Which you might want to take into consideration if you're wanting to de-prioritize something.

I suspect that some grouping/spacing of related icons might be helpful but I suspect I'm heading into bike-shedding territory without any data to back that up. :) e.g.

```
[ A ][ B ][ C ] [ D ][ E ] [ F ][ G ]
```

**Comment by Sawuare** [comment]
I'm convinced. Mixer before controller rack is better than the other way around. I edited the issue and I'll change #3938 accordingly.

**Comment by Sawuare** [comment]
Closed via 0d73f32.
